specs/protocol/tx_format.md

```diff
@@ -239,13 +241,40 @@ Note: when verifying a predicate, `txID`, `outputIndex`, `balanceRoot`, `stateRo
 
 Note: when executing a script, `txID`, `outputIndex`, `balanceRoot`, and `stateRoot` are initialized to the transaction ID, output index, amount, and state root of the contract with ID `contractID`, and `txPointer` is initialized to zero.
 
+### InputMessage
+
+| name                  | type        | description                                                            |
+|-----------------------|-------------|------------------------------------------------------------------------|
+| `messageID`           | `byte[32]`  | The messageID as described [here](./identifiers.md#message-id).        |
+| `sender`              | `byte[32]`  | The address of the message sender.                                     |
+| `recipient`           | `byte[32]`  | The address of the message recipient.                                  |
+| `dataLength`          | `uint16`    | Length of message data, in bytes.                                      |
+| `data`                | `byte[]`    | The message data or [abi encoded](./abi.md) call to execute.           |
```

Comment:
Move this to the back. Note that fixed-length field are first, then all variable-length fields. This is required to be able to read fixed-length fields in constant time.